### ISSUE TITLE
Provide constants for e.g. the zero address and max uint256

### ISSUE THREAD

**Issue by KaiRo-at** [ethereum/web3.py#2095]
* Version: 5.22.0
* Python: 3.8
* OS: linux

### What was wrong?

If I need to compare an address returned from a contract or variable to the zero address or want to e.g. use the maximum allowed uint256, I need to define those myself.

### How can it be fixed?

Provide values like this as constants from Web3.py - see also what Ethers has available at https://docs.ethers.io/v5/api/utils/constants/ - for me, the zero address (needed often) and the max uint256 (sometimes needed to e.g. set a maximum allowance on ERC20) are the most important ones, I also sometimes have use cases for a full 32byte hex value of zero, which is "HashZero" in the Ethers list. The wei per ether can be retrieved via `Web3.toWei(1, 'ether')` but may be nice, the others in the Ethers list I think are not very useful in Python.

**Comment by kclowes** [comment]
Sure, I like that idea. On first glance, what I would do is make a new file called `constants.py` in the `web3/` dir and then go search the codebase for the hard-coded constants and replace them. Docs should be added too.

**Comment by KaiRo-at** [comment]
Sounds great!
I was looking at code my colleagues did with Ethers and found there's a constant there for the zero address, and told them to use that, and then thought about how I always define this in my Python code somewhere and wondered if Web3.py has that to refer to as well, so I looked for it and when I didn't find it, I thought I probably should file this as an enhancement request. ;-)

**Comment by ernestosperanza** [comment]
Hello, I'm interested in working on this issue and contribute to the project, let me know if I can help.
Thanks!

**Comment by smishy05** [comment]
Hey there! I am interested in working on this issue. Please let me know if I can contribute.
Many thanks!

**Comment by kclowes** [comment]
It looks like @ernestosperanza got there first, @smishy05. But if there isn't a PR in the next few days, feel free to pick it up. Thanks for the interest!

**Comment by ernestosperanza** [comment]
Yes sorry, it's my first issue. It might take some time but I am working on it.

**Comment by smishy05** [comment]
@kclowes Thanks for the reply! Sounds fair enough!

@ernestosperanza No issues! Good luck!
